internal/pkg/instrumentation/probe/probe.go

```diff
@@ -365,50 +371,76 @@ type Uprobe struct {
 	// function specified by Sym. If ReturnProbe is empty, no eBPF program will be attached to the return of the function.
 	ReturnProbe string
 	DependsOn   []string
+
+	closers atomic.Pointer[[]io.Closer]
```

Comment:
Why not `[]io.Closer`?
What are the concurrent scenarios we are trying to protect against?

Reply:
The `Uprobe.Close` is called by the `Manager` which can shutdown asynchronously. For example, this can occur when the context passed to the new `Instrumentation` is canceled asynchronously[^1]. This is to prevent that data-race.

[^1]: https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/aeb95422568a2e2901a02620527e526161936321/cli/main.go#L87-L90

Reply:
I see. Do you think we can use `sync.Once` instead? I think it might be more readable